tests/staticFilesServer.test.js

```diff
@@ -0,0 +1,158 @@
+  })}px;
+    }
+
+    .${faker.lorem.word()} {
+      background-color: ${faker.internet.color()};
+      border: 2px solid ${faker.internet.color()};
+      padding: ${faker.number.int({
+    min: 5, max: 20,
+  })}px;
+    }
+  `;
+}
+
+describe('Static files server', () => {
+  describe('createServer', () => {
+    describe('basic scenarios', () => {
+      it('should create a server', () => {
+        expect(createServer)
+          .toBeInstanceOf(Function);
+      });
+
+      it('should create an instance of Server', async() => {
+        expect(createServer())
+          .toBeInstanceOf(Server);
+      });
+    });
+
+    describe('Server', () => {
+      let server;
+      const publikFolderPath = path.resolve(__dirname, '../public');
+      const stylesFolderPath = path.resolve(__dirname, '../public/styles');
+      const indexFilePath = path.resolve(__dirname, '../public/index.html');
+      const mainCSSFilePath = path.resolve(__dirname, '../public/styles/main.css');
+      const randomCSSContent = generateRandomCSS();
+
+      beforeEach(() => {
+        server = createServer();
+        server.listen(PORT);
+
+        if (!fs.existsSync(publikFolderPath)) {
+          fs.mkdirSync(publikFolderPath);
+        }
+
+        if (!fs.existsSync(stylesFolderPath)) {
+          fs.mkdirSync(stylesFolderPath);
+        }
+
+        if (!fs.existsSync(indexFilePath)) {
+          fs.writeFileSync(indexFilePath, '<!DOCTYPE html>');
+        }
+
+        if (!fs.existsSync(mainCSSFilePath)) {
+          fs.writeFileSync(mainCSSFilePath, randomCSSContent);
+        }
+      });
+
+      afterEach(() => {
+        server.close();
+
+        if (fs.existsSync(publikFolderPath)) {
+          fs.rmdirSync(publikFolderPath, { recursive: true });
+        }
+
+        if (fs.existsSync(stylesFolderPath)) {
+          fs.rmdirSync(stylesFolderPath, { recursive: true });
+        }
+      });
+
+      describe('Valid file requests', () => {
+        it('should return the correct file for a valid path', async() => {
+          const response = await axios.get(`${HOST}/file/index.html`);
+
+          expect(response.status).toBe(200);
+          expect(response.data).toContain('<!DOCTYPE html>');
+        });
+
+        it('should return the correct file for a valid subfolder path', async() => {
+          const response = await axios.get(`${HOST}/file/styles/main.css`);
+
+          expect(response.status).toBe(200);
+          expect(response.data).toContain(randomCSSContent);
+        });
+      });
+
+      describe('Non-existent file requests', () => {
+        it('should return 404 for non-existent files', async() => {
+          try {
+            await axios.get(`${HOST}/file/nonexistentfile.txt`);
+          } catch (error) {
+            expect(error.response.status).toBe(404);
+            expect(error.response.headers['content-type']).toBe('text/plain');
+            expect(error.response.data.length).toBeGreaterThan(0);
+          }
```

Comment:
There is not guarantee that catch will be executed 
And the test will just end
You need to check the number of assertions
https://jestjs.io/docs/expect#expectassertionsnumber

Correct me please if I'm wrong 
And this approach fails in case of success response

Reply:
good point here, thanks fixed)